include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -1426,6 +1434,72 @@ inline Value Value::CreateOpaque(const char* domain, const char* type_name, cons
 //
 // Custom OP Inlines
 //
+inline Logger::Logger(const OrtLogger* logger) : logger_(logger) {
+  Ort::ThrowOnError(GetApi().Logger_GetLoggingSeverityLevel(this->logger_, &this->cached_severity_level_));
+}
+
+inline OrtLoggingLevel Logger::GetLoggingSeverityLevel() const noexcept {
+  return cached_severity_level_;
+}
+
+inline Status Logger::LogMessage(OrtLoggingLevel log_severity_level, const ORTCHAR_T* file_path, int line_number,
+                                 const char* func_name, const char* message) const noexcept {
+  OrtStatus* status = GetApi().Logger_LogMessage(logger_, log_severity_level, message, file_path, line_number,
+                                                 func_name);
+  return Status{status};
+}
+
+// Disable warnings about the format string not being a literal (-Wformat-nonliteral and -Wformat-security)
+// for gcc and clang. The alternative is to use actual C-style variadic parameters and apply
+// __attribute__(format(printf...)), which does not work with variadic templates.
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#pragma GCC diagnostic ignored "-Wformat-security"
+#elif defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
+#pragma clang diagnostic ignored "-Wformat-security"
+#endif
+template <typename... Args>
+inline Status Logger::LogFormattedMessage(OrtLoggingLevel log_severity_level, const ORTCHAR_T* file_path,
+                                          int line_number, const char* func_name, const char* format,
+                                          Args&&... args) const noexcept {
+  int msg_len = snprintf(nullptr, 0U, format, std::forward<Args>(args)...);
```

Comment:
>snprintf [](http://example.com/codeflow?start=16&length=8)

snprintf requires #include <stdio.h>

Reply:
I'm including `<cstdio>` in onnxruntime_cxx_api.h.

Reply:
> I'm including `<cstdio>` in onnxruntime_cxx_api.h.

That would be `std::snprintf`.

Reply:
Ah yes, you're right. I updated to use `std::`. 

Note: It looks like using just `snprintf` with `<cstdio>` works for most compilers, but it is [not guaranteed to work across all implementations](https://stackoverflow.com/a/29900474).